### PR TITLE
feat: Added a Generator producing Characters

### PR DESCRIPTION
**Pull request by Kantis**
As suggested in issue #84 

`StringCharSets` renamed to simply `CharSets` and moved into the `Characters.kt` file instead.

String generators were refactored to use a character generator instead.

### CODE REVIEW

**Comment by jcornaz** [comment]
Thanks very much @Kantis!

I'll try to look at it this week.

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt [hunk 1]
```diff
@@ -10,84 +10,45 @@ import kotlin.random.Random
 /**
  * Returns a generator of String of length between [minLength] and [maxLength] (inclusive)
  *
- * @param charset Set of character to be used in generated strings
- * @param exclude Characters to exclude from generated strings
+ * @param charGenerator A generator of characters which will be used to construct the strings
  */
 fun Generator.Companion.strings(
```
Comment (jcornaz) [inline generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:15]: Can you leave the old function signatures? Their implementation should be a simple delegation to the new ones:

Example: 

```kotlin
fun Generator.Companion.strings(
  minLength: Int = 0,
  maxLength: Int = maxOf(minLength, KWIK_DEFAULT_MAX_SIZE),
  charset: Set<Char> = CharSets.printable,
  exclude: Set<Char> = emptySet(),
) = strings(minLength, maxLength, characters(charset, exclude))
```

Maybe I'll deprecate them or maybe I'll keep the overlaod. But either way it is nice to not *force* the user to use the new API right away.

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt [hunk 2]
```diff
@@ -10,84 +10,45 @@ import kotlin.random.Random
...
 fun Generator.Companion.nonBlankStrings(
     maxLength: Int = KWIK_DEFAULT_MAX_SIZE,
-    charset: Set<Char> = StringCharSets.printable,
-    exclude: Set<Char> = emptySet()
-): Generator<String> = nonEmptyStrings(maxLength, charset, exclude).filterNot { it.isBlank() }
-
-/**
- * Common set of character to be used with string generators
- */
-object StringCharSets {
```
Comment (jcornaz) [inline generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:69]: Please leave it, and only add a deprecation annotation (`@Deprecated`).

**Review by jcornaz — CHANGES_REQUESTED** [review]
Looks great thanks!

In the current form it would be a breaking change. But I think it is quite easy to make it non-breaking.

You can simply leave the old API declaration, and let it delegate to the new functions. That way this PR is only *adding* a new generator and overloads for the string generator.

For the StringCharset, it can be kept as deprecated. ;-)

Thanks for opening a great PR that contains documentation update and tests.

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt (hunk 1 above)
Comment (Kantis) [reply generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:15]: Thanks, I completely forgot about backwards compatability. I will restore the signatures. Should I restore the old tests as well?

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt (hunk 2 above)
Comment (Kantis) [reply generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:69]: Restored it, along with its' tests

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt [hunk 3]
```diff
@@ -13,29 +13,39 @@ import kotlin.random.Random
  * @param charset Set of character to be used in generated strings
  * @param exclude Characters to exclude from generated strings
  */
+@Deprecated(
+    "Use `strings(minLength, maxLength, charGenerator)` instead",
+    ReplaceWith("strings(minLength, maxLength, charGenerator)")
+)
```
Comment (jcornaz) [inline generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:19]: I think I'd like to keep this overload. At least for now. 
```suggestion
```

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt [hunk 4]
```diff
@@ -45,27 +55,58 @@ fun Generator.Companion.strings(
  * @param charset Set of character to be used in generated strings
  * @param exclude Characters to exclude from generated strings
  */
+@Deprecated(
+    "Use `nonEmptyStrings(maxLength, charGenerator)` instead",
+    ReplaceWith("nonEmptyStrings(maxLength, charGenerator)")
+)
```
Comment (jcornaz) [inline generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:61]: I think I'd like to keep this overload. At least for now.

```suggestion
@Deprecated(
    "Use `nonEmptyStrings(maxLength, charGenerator)` instead",
    ReplaceWith("nonEmptyStrings(maxLength, charGenerator)")
)
```

File: generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt [hunk 5]
```diff
@@ -45,27 +55,58 @@ fun Generator.Companion.strings(
...
 /**
  * Returns a generator of non-blank String of length between 1 and [maxLength] (inclusive)
  *
  * @param charset Set of character to be used in generated strings
  * @param exclude Characters to exclude from generated strings
  */
+@Deprecated(
+    "Use `nonBlankStrings(maxLength, charGenerator)` instead",
+    ReplaceWith("nonEmptyStrings(maxLength, charGenerator)")
+)
```
Comment (jcornaz) [inline generator/stdlib/src/commonMain/kotlin/com/github/jcornaz/kwik/generator/stdlib/Strings.kt:88]: I think I'd like to keep this overload. At least for now.

```suggestion
```

**Review by jcornaz — APPROVED** [review]
Great pull request thanks!

**Pull request merged**